src/parsimmon.js

```diff
@@ -105,19 +105,43 @@
 
   // For ensuring we have the right argument types
   function assertParser(p) {
-    if (!(p instanceof Parser)) throw new Error('not a parser: '+p);
+    if (!(p instanceof Parser)) {
+      throw new Error('not a parser: ' + p);
+    }
   }
+
   function assertNumber(x) {
-    if (typeof x !== 'number') throw new Error('not a number: '+x);
+    if (typeof x !== 'number') {
+      throw new Error('not a number: ' + x);
+    }
   }
+
   function assertRegexp(x) {
-    if (!(x instanceof RegExp)) throw new Error('not a regex: '+x);
+    if (!(x instanceof RegExp)) {
+      throw new Error('not a regex: '+x);
+    }
+    var f = flags(x);
+    for (var i = 0; i < f.length; i++) {
+      var c = f.charAt(i);
+      // Only allow regexp flags [imu] for now, since [g] and [y] specifically
+      // mess up Parsimmon. If more non-stateful regexp flags are added in the
+      // future, this will need to be revisited.
+      if (c != 'i' && c != 'm' && c != 'u') {
```

Comment:
> If more non-stateful regexp flags are added in the future, this will need to be revisited.

Which is why we should blacklist `g` and `y` here instead of whitelisting `i`, `m`, and `u`.